File: examples/python/break_test.py
import time
import rby1_sdk
from rby1_sdk import *
import argparse
import numpy as np


def main(address, joint):
    robot = rby1_sdk.create_robot_a(address)
    robot.connect()

    if not robot.is_connected():
        print("Robot is not connected")
        exit(1)

    if not robot.is_power_on(".*"):
        rv = robot.power_on(".*")
        if not rv:
            print("Failed to power on")
            exit(1)

    time.sleep(0.5)
    print("Break Release!")
    if not robot.break_release(joint):
        print("Error: Failed to break release.")

    time.sleep(0.5)
    print("Break Engage!")
    if not robot.break_engage(joint):
        print("Error: Failed to break engage.")


if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="break test")
    parser.add_argument("--address", type=str, required=True, help="Robot address")
    parser.add_argument("--joint", type=str, help="Joint name regex pattern")

    args = parser.parse_args()

    if "torso" in args.joint or args.joint == ".*":
        print(f"Warning: Using {args.joint} may cause the robot to collapse.")
    else:
        main(address=args.address, joint=args.joint)

File: examples/cpp/break_test.cpp
#include <chrono>
#include <iomanip>
#include <iostream>
#include <thread>
#include "rby1-sdk/model.h"
#include "rby1-sdk/robot.h"

using namespace rb;
using namespace std::chrono_literals;

int main(int argc, char** argv) {
  if (argc < 3) {
    std::cerr << "Usage: " << argv[0] << " <server address> <joint name>" << std::endl;
    return 1;
  }

  std::string address{argv[1]};
  std::string joint_name{argv[2]};

  if (joint_name.find("torso") != std::string::npos || joint_name == ".*") {
    std::cerr << "Warning: Using '" << joint_name << "' may cause the robot to collapse." << std::endl;
    return 1;
  }

  auto robot = Robot<y1_model::A>::Create(address);

  if (!robot->Connect()) {
    std::cerr << "Connection failed" << std::endl;
    return 1;
  }

  robot->PowerOn(".*");
  std::this_thread::sleep_for(0.5s);
  std::cout << "Break Release!" << std::endl;
  if (!robot->BreakRelease(joint_name)) {
    std::cerr << "Error: Failed to break release." << std::endl;
    return 1;
  }
  std::this_thread::sleep_for(0.5s);
  std::cout << "Break Engage!" << std::endl;
  robot->BreakEngage(joint_name);
  if (!robot->BreakEngage(joint_name)) {
    std::cerr << "Error: Failed to break engage." << std::endl;
    return 1;
  }

  return 0;
}